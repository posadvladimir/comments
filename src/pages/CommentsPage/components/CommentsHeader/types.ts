
export type TCommentsHeaderProps = {
    totalComments: number;
    likes: number;
}