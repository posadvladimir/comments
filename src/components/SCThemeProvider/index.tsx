
import { ThemeProvider } from 'styled-components';

import type { TChildrenProps } from 'src/ts/types/children';

import theme from 'src/styles/theme';

const SCThemeProvider: React.FC<TChildrenProps> = ({ children }) => (
    <ThemeProvider theme={theme}>
        {children}
    </ThemeProvider>
);

export default SCThemeProvider;