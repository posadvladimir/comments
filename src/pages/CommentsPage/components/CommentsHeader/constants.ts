
import type { TDeclensionArray } from '../../common/ts/types/declensions';

export const COMMENT_DECLENSIONS: TDeclensionArray = [
    'комментарий',
    'комментария',
    'комментариев'
];