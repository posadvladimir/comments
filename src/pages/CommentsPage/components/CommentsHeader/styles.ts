
import { styled } from 'styled-components';

export const CommentsHeaderWrapper = styled.header`
    display: flex;
    justify-content: space-between;
    padding-bottom: 10px;
    border-bottom: 0.2px solid rgba(118, 118, 118, 0.4);
`;