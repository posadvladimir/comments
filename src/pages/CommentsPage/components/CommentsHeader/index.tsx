
import type { TCommentsHeaderProps } from './types';

import { LikeIconStates } from '../../common/ts/enums/likeIconStates';

import { COMMENT_DECLENSIONS } from './constants';

import { getRuNumberFormat } from '../../common/helpers/getRuNumberFormat';
import { getWordDeclension } from '../../common/helpers/getWordDeclension';

import { BoldSpan } from '../../common/styles/boldSpan';
import * as S from './styles';

import LikeCounter from '../LikeCounter';

const CommentsHeader: React.FC<TCommentsHeaderProps> = ({
    totalComments,
    likes,
}) => (
    <>
        <S.CommentsHeaderWrapper>
            <BoldSpan>{getRuNumberFormat(totalComments)} {getWordDeclension(totalComments, COMMENT_DECLENSIONS)}</BoldSpan>
            <LikeCounter
                likes={likes}
                iconState={LikeIconStates.GreyStroke}
                disabled
            />
        </S.CommentsHeaderWrapper>
    </>
);

export default CommentsHeader;