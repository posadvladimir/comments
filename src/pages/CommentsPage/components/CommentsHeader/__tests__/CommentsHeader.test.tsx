
import { render, screen } from '@testing-library/react';

import { runSnapshotTest } from 'src/common/tests/snapshot';

import SCThemeProvider from 'src/components/SCThemeProvider';
import CommentsHeader from '..';

const COMMENTS_TOTAL = 5000;
const LIKES_TOTAL = 1000;

const commentsHeaderComponent = (
    <SCThemeProvider>
        <CommentsHeader
            totalComments={COMMENTS_TOTAL}
            likes={LIKES_TOTAL}
        />
    </SCThemeProvider>
);

const renderCommentsHeader = (): void => {
    render(commentsHeaderComponent);
};

describe('CommentsHeader', () => {
    test('отображает количество комментариев в правильном формате', () => {    
        renderCommentsHeader();

        const formattedComments = screen.getByText('5 000 комментариев');

        expect(formattedComments).toBeInTheDocument();
    });

    test('отображает количество лайков в правильном формате', () => {    
        renderCommentsHeader();
    
        const formattedLikes = screen.getByText('1 000');
    
        expect(formattedLikes).toBeInTheDocument();
    });

    runSnapshotTest(commentsHeaderComponent);
});