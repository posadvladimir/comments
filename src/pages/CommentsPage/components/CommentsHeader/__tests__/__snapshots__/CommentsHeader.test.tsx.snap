// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CommentsHeader соотвествует снепшоту 1`] = `
<header
  className="sc-grXZZQ krEezl"
>
  <span
    className="sc-bcPKhP jtwZFa"
  >
    5 000
     
    комментариев
  </span>
  <div
    className="sc-dkjKgF bbldwS"
    data-testid="like-counter-test-id"
  >
    <button
      className="sc-hLclGa deDrpk"
      data-testid="like-counter-button-test-id"
      disabled={true}
      onClick={[Function]}
    >
      <img
        alt="like icon"
        className="sc-eDnVMP mhaLA"
        src="grey-stroke.png"
      />
    </button>
    <span
      className="sc-jSwlEQ hgxKmw"
      data-testid="like-counter-total-test-id"
    >
      1 000
    </span>
  </div>
</header>
`;
